pkg/appgw/ingress_rules.go

```diff
@@ -14,23 +14,46 @@ import (
 	"github.com/Azure/application-gateway-kubernetes-ingress/pkg/environment"
 )
 
-func (c *appGwConfigBuilder) getListenersFromIngress(ingress *v1beta1.Ingress, env environment.EnvVariables) map[listenerIdentifier]listenerAzConfig {
+func (c *appGwConfigBuilder) getListenersFromIngress(ingress *v1beta1.Ingress, env environment.EnvVariables, ingressWafPolicy map[*v1beta1.Ingress]map[string]bool) map[listenerIdentifier]listenerAzConfig {
 	listeners := make(map[listenerIdentifier]listenerAzConfig)
+
+	// if ingress has only backend configured
+	if ingress.Spec.Backend != nil && len(ingress.Spec.Rules) == 0 {
+		return listeners
+	}
+
 	for ruleIdx := range ingress.Spec.Rules {
 		rule := &ingress.Spec.Rules[ruleIdx]
 		if rule.HTTP == nil {
 			continue
 		}
 
-		_, ruleListeners := c.processIngressRule(rule, ingress, env)
+		_, ruleListeners := c.processIngressRuleWithTLS(rule, ingress, env)
+
 		for k, v := range ruleListeners {
+			if ingressWafPolicy != nil && ingressWafPolicy[ingress][rule.Host] {
+				c.processIngresswithWafPolicy(&v, ingress, env)
+			}
 			listeners[k] = v
 		}
 	}
+
 	return listeners
 }
 
-func (c *appGwConfigBuilder) processIngressRule(rule *v1beta1.IngressRule, ingress *v1beta1.Ingress, env environment.EnvVariables) (map[Port]interface{}, map[listenerIdentifier]listenerAzConfig) {
+func (c *appGwConfigBuilder) processIngresswithWafPolicy(azConfig *listenerAzConfig, ingress *v1beta1.Ingress, env environment.EnvVariables) {
+	policy, _ := annotations.WAFPolicy(ingress)
+	if policy != "" {
+		if env.AttachWAFPolicyToListener {
+			// logging to see if customer configures env.AttachWAFPolicyToListener or not
+			glog.V(5).Info("AttachWAFPolicyToListener is enabled")
+		}
+		glog.V(5).Infof("Found WAF policy: %s in annotation", policy)
```

Comment:
glog.V(5).Infof("Attaching WAF policy: %s to listener <> ?", policy)
Do we need to add info about the listener as well ? Maybe the host name ?